pandas/core/indexes/numeric.py

```diff
@@ -321,13 +341,22 @@ class Float64Index(NumericIndex):
 
     _typ = 'float64index'
     _engine_type = libindex.Float64Engine
-    _left_indexer_unique = libjoin.left_join_indexer_unique_float64
-    _left_indexer = libjoin.left_join_indexer_float64
-    _inner_indexer = libjoin.inner_join_indexer_float64
-    _outer_indexer = libjoin.outer_join_indexer_float64
-
     _default_dtype = np.float64
 
+    # Cython methods; see github.com/cython/cython/issues/2647
+    #  for why we need to wrap these instead of making them class attributes
+    def _left_indexer_unique(self, left, right):
```

Comment:
we could think about doing generic dispatch in the base class,
e.g.
```
def _left_indexer(self, left, right):
    return getattr(libjoin, 'left_join_indexer_{}.format(self._join_type))(left, right)
```

might be slightly less readable, but less code overall in the subclasses.

Reply:
Actually we can do one better.  By just calling `left_join_indexer` cython will dispatch to the appropriate version on its own.  (according to the docs; I haven't tried this myself)